agent/agent.go

```diff
@@ -345,6 +363,18 @@ func (a *Agent) connectionManager(ctx context.Context, cancel context.CancelFunc
 		a.connectedMu.Unlock()
 
 		go a.receiveLoop(ctx, cancel, conn)
```

Comment:
Don't we need to start the `receiveLoop` after the first message from the server has been received?

Reply:
As discussed, `receiveLoop` needs to be started in order to receive the entity config update, but we only add the check request handler once we have the entity config